src/main/java/games/strategy/engine/lobby/server/userDB/DbUser.java

```diff
@@ -1,42 +1,113 @@
 package games.strategy.engine.lobby.server.userDB;
 
-import java.util.Date;
+import com.google.common.base.Preconditions;
 
+import games.strategy.engine.framework.startup.ui.InGameLobbyWatcher;
+import games.strategy.util.Util;
+
+/**
+ * Value object representing data needed to create a user in database.
+ */
 public class DbUser {
-  private final String name;
-  private final String email;
-  private final boolean isAdmin;
-  private final Date lastLogin;
-  private final Date joined;
+  public final String name;
+  public final String email;
+  public final boolean admin;
+
+  public enum Role {
+    NOT_ADMIN, ADMIN
+  }
+
+  public static class UserName {
```

Comment:
It avoids the 'Stringly' typed values we had before. Instead of "new User(String,String,String)", we actually get some types "new User(User.UserName, User.UserEmail)"

A builder type pattern is something of an alternative, it would make these value types less useful.

For context, the value in these types has been shown while I was working on this PR. The HashPassword was split off to its own class when previously it was part of DbUser. That provided good flexibiilty and the updates to do that were minimal. We also had opportunity to consolidate some validation logic in that class, and it was suddenly much more testable as well.